src/ERC4973Permit.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity ^0.8.6;
+
+import "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+
+abstract contract ERC4973Permit is EIP712 {
+    error ErrorUnauthorizedMinting();
+
+    bytes32 private constant MINT_PERMIT_TYPEHASH =
+        keccak256(
+            "MintPermit(address from,address to,string tokenURI)"
+        );
+
+    constructor(string memory name) EIP712(name, "1") {}
```

Comment:
Make "1" parameter to EIP712 configurable through ERC4973Permit constructor as is done with `name`